testnet/stacks-node/src/neon_node.rs

```diff
@@ -261,7 +265,7 @@ fn spawn_peer(mut this: PeerNetwork, p2p_sock: &SocketAddr, rpc_sock: &SocketAdd
             let network_result = this.run(&burndb, &mut chainstate, &mut mem_pool, Some(&mut dns_client), download_backpressure, poll_ms)
                 .unwrap();
 
-            if network_result.has_blocks() || network_result.has_microblocks() {
+            if network_result.has_data_to_store() {
```

Comment:
Wait.. :)
![Screenshot from 2020-06-03 16-34-10](https://user-images.githubusercontent.com/87777/83686418-2a45bb80-a5b8-11ea-9b2f-7e6be73ac27c.png)

Reply:
Yes -- I wanted to revisit that discussion :)

Notably -- the integration test here _will not_ pass unless this line is changed.

Reply:
I reverted it in my other PR.  If it's causing breakage here, then it can be reverted (my rationale was only for improving performance).